File: java/binary_search.java
public class binary_search {  
 public static int binarySearch(int arr[], int val){
    int l = 0, r = arr.length - 1;
    while(l <= r){  
      int mid = (l + r) >> 1;  
      if (arr[mid] == val){  
        return mid;  
      } else if(arr[mid] < val){  
        l = mid + 1;     
      }else{  
         r = mid - 1;  
      }
   }
   return -1;
 }
 public static void main(String args[]){  
    int arr[] = {10,20,30,40,50};  
    int val = 30;  
    int ans = binarySearch(arr,val);
    System.out.println((ans == -1) ? "Element not found" : "Element is found at index: " + ans);  
 }
}

File: rust/binary_search.rs
fn binary_search(arr: &mut [i32], n: i32) -> i32 {
    let mut start = 0;
    let mut end = arr.len();
    arr.sort();
    let mut mid;
    let mut res: i32 = -1;
    while start <= end {
        mid = (start + end) / 2;
        if arr[mid] == n {
            res = mid as i32;
            break
        } else if arr[mid] < n {
            start = mid + 1;
        } else { 
            end = mid - 1;
        }
    }
    return res
}

fn main () {
  let mut A = [1,2,3,4,5,6,7,8,9, 10];
  let n = binary_search(&mut A, 5);
  println!("{}", n);
}
